ocaml/xapi/sm_exec.ml

```diff
@@ -265,6 +265,7 @@ let parse_attach_result (xml : Xml.xml) =
   rethrow_parse_failures (Xml.to_string_fmt xml) (fun () ->
       let info = XMLRPC.From.structure xml in
       let params = XMLRPC.From.string (safe_assoc "params" info) in
+      let params_nbd = XMLRPC.From.string (safe_assoc "params_nbd" info) in
```

Comment:
It this new param now mandatory? This code raises an error if it is absent.

Reply:
The param_nbd is  always be passed back from driver/blktap2.py see 
https://github.com/xapi-project/sm/pull/482 

This commit is dangerous if not taken in conjunction with the sm changes.

I don't think there is any way this code would be called in a none SMAP V1 context, but perhaps @edwintorok  can confirm

Reply:
To make it less dangerous, and the type is changed to an option type, then we can just set the value here to `None` is the key is missing, unless it is important to report this as an error right here.

Reply:
Great, That will make integration simpler with SM